client/pages/vehicle-alerts.jsx

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+
+export default class VehicleAlerts extends React.Component {
+  constructor(props) {
+    super(props);
+    const vId = 2;// Replace this when creating vehicle list feature
+    this.state = { vehicle: {}, vehicleId: vId, demoVehicle: false };
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.sendText = this.sendText.bind(this);
+    this.updateAlerts = this.updateAlerts.bind(this);
+
+    fetch(`/api/vehicleinfo/${vId}`)
+      .then(res => res.json())
+      .then(result => {
+        this.setState({ vehicle: result });
+      });
+
+  }
+
+  sendText(text) {
+    fetch('/api/sendtext/', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+
+      body: JSON.stringify(text)
+    })
+      .then(response => response.json())
+      .catch(error => {
+        console.error('Error:', error);
+      });
+  }
+
+  updateAlerts(alerts) {
+    fetch(`/api/vehiclealerts/${this.state.vehicleId}`, {
+      method: 'PUT',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(alerts)
+    })
+      .then(response => response.json())
+      .catch(error => {
+        console.error('Error:', error);
+      });
+  }
+
+  handleSubmit(event) {
+
+    const alerts = { speedAlert: event.target[0].checked, speedingThreshold: event.target[1].value, brakeAlert: event.target[2].checked, accelerationAlert: event.target[3].checked, textAlert: event.target[4].checked, emailAlert: event.target[5].checked };
+    this.setState({ vehicle: alerts });
+    if (this.state.vehicle.demo) {
+      if (alerts.textAlert) {
+        if (this.state.vehicle.topSpeed > alerts.speedingThreshold) {
+          const message = `${this.state.vehicle.name} has exceeded speeding threshold. Speed: ${alerts.topSpeed}`;
+          const text = { toNumber: '+19492664664', message: message };
+          this.sendText(text);
+        }
+        if (alert.emailAlert) {
+          // email alert code goes here. -make function
+        }
+      }
+
+    }
+
+    this.updateAlerts(alerts);
+
+    event.preventDefault();
+  }
+
+  render() {
+
+    const currentVehicle = Object.assign({}, this.state.vehicle);
```

Comment:
Copying your state here is unnecessary. You only want to copy state if you are going to make modifications to it inside of an event handler or data fetching.